### PR TITLE
Add custom aggrid

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/components/OverflowableText/overflowable-text.jsx [hunk 1]
```diff
@@ -15,6 +15,8 @@ const overflowStyle = {
         whiteSpace: 'pre',
         textOverflow: 'ellipsis',
         overflow: 'hidden',
+        lineHeight: 'initial',
+        verticalAlign: 'middle',
```
Comment (klesaulnier) [inline src/components/OverflowableText/overflowable-text.jsx:19]: make a wrapper instead of customizing the very generic component

**Comment by klesaulnier** [comment]
Remember to remove everything linked to sorting/filtering in react-virtualized - I don't think it's used anymore after the changes in gridexplore

File: src/components/OverflowableText/overflowable-text.jsx (hunk 1 above)
Comment (Meklo) [reply src/components/OverflowableText/overflowable-text.jsx]: I found a way to stylize it on gridexplore side

**Comment by Meklo** [comment]
> Remember to remove everything linked to sorting/filtering in react-virtualized - I don't think it's used anymore after the changes in gridexplore

The file containing the sort logic is tightly coupled to MUIVirtualizedTable, maybe it's more judicious to remove it altogether once we get rid of MUIVirtualizedTable

File: src/components/CustomAGGrid/custom-aggrid.style.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (Meklo) [reply src/components/CustomAGGrid/custom-aggrid.style.ts]: Done

File: src/components/CustomAGGrid/custom-aggrid.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (Meklo) [reply src/components/CustomAGGrid/custom-aggrid.tsx]: Done

File: src/components/CustomAGGrid/custom-aggrid.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    const colDefinedMinWidth = column?.getColDef()?.minWidth;
+    if (column && colDefinedMinWidth && finished) {
+        const newWidth = column?.getActualWidth();
+        if (newWidth < colDefinedMinWidth) {
+            column?.setActualWidth(colDefinedMinWidth, params.source);
+        }
+    }
+};
+
+const CustomAGGrid = React.forwardRef<any, CustomAGGridProps>((props, ref) => {
```
[the remark this answers is not in the document]
Comment (Meklo) [reply src/components/CustomAGGrid/custom-aggrid.tsx]: Done

File: src/module-mui.d.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (Meklo) [reply src/module-mui.d.ts]: Done

File: src/module-mui.d.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// used to customize mui theme
+// https://mui.com/material-ui/customization/theming/#typescript
+declare module '@mui/material/styles/createTheme' {
+    interface Theme {
```
[the remark this answers is not in the document]
Comment (Meklo) [reply src/module-mui.d.ts:11]: Done

**Review by klesaulnier — APPROVED** [review]

**Review by achour94 — APPROVED** [review]

**Pull request merged**